tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_beacon.c

```diff
@@ -808,6 +810,111 @@ static void test_rx_multiple_netkeys(void)
 
 }
 
+static struct k_work_delayable beacon_timer;
+
+static void secure_beacon_send(struct k_work *work)
+{
+	NET_BUF_SIMPLE_DEFINE(buf, 22);
+	beacon_create(&buf, test_net_key, 0, 0);
+	send_beacon(&buf);
+	/**
+	 * Sending SNB(secure network beacon) faster to guarantee
+	 * at least one beacon is received by tx node in 10s period.
+	 */
+	k_work_schedule(&beacon_timer, K_SECONDS(4));
+}
+
+/* Test beacon interval. */
+static void test_tx_secure_beacon(void)
+{
+	/* initial SNB interval is 10s*/
+	int64_t min_beacon_interval = 10*MSEC_PER_SEC; /* ms */
+	int64_t max_beacon_interval = 10*MSEC_PER_SEC; /* ms */
+	int64_t send_time;
+	int64_t delta;
+	int sent_beacon_nbr = 0;
+	struct bt_mesh_subnet *sub;
+
+	bt_mesh_test_cfg_set(&tx_cfg, 680);
+	k_sleep(K_SECONDS(2));
+	bt_mesh_test_setup();
+
+	sub = bt_mesh_subnet_find(NULL, NULL);
+	send_time = (int64_t)sub->beacon_sent;
+
+	/* count the beacons send and verify that the SNB intervals are in the accepted range. */
+	while (sent_beacon_nbr < 5) {
+		if (send_time != sub->beacon_sent) {
+			delta = sub->beacon_sent - send_time;
+			if (delta < min_beacon_interval) {
+				min_beacon_interval = delta;
+			} else if (delta > max_beacon_interval) {
+				max_beacon_interval = delta;
+			}
+			send_time = sub->beacon_sent;
+			sent_beacon_nbr++;
+		}
+		k_sleep(K_SECONDS(2));
+	}
+
+	ASSERT_TRUE(min_beacon_interval >= (10*MSEC_PER_SEC));
+	ASSERT_TRUE(max_beacon_interval <= (600*MSEC_PER_SEC) + 100);
+	PASS();
+}
+
+static void test_rx_secure_beacon(void)
+{
+	NET_BUF_SIMPLE_DEFINE(buf, 22);
+	int err;
+	int64_t beacon_recv_time;
+	int64_t delta;
+
+	bt_mesh_test_cfg_set(&rx_cfg, 680);
+	k_sem_init(&observer_sem, 0, 1);
+	k_work_init_delayable(&beacon_timer, secure_beacon_send);
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Bluetooth init failed (err %d)", err);
+	}
+
+	beacon_create(&buf, test_net_key, 0, 0);
+	k_sleep(K_SECONDS(5));
+	/*wait provisioned tx node to send the first beacon*/
+	ASSERT_TRUE(wait_for_beacon(NULL, NULL));
+	k_sleep(K_SECONDS(2));
+
+	/*send out SNB with 20ms interval and observe tx node also sends in 20ms, too */
+	for (size_t i = 0; i < 4; i++) {
+		if (i % 2) {
+			send_beacon(&buf);
+			ASSERT_FALSE(wait_for_beacon(NULL, NULL));
+		} else {
+			ASSERT_TRUE(wait_for_beacon(NULL, NULL));
+		}
+	}
+
+	/* wait for a SNB from tx node */
```

Comment:
What does prevent to sneak the fifth beacon before line 898?

Reply:
Can you elaborate the question, please?

Reply:
I mean time rally. What happens if the fifth (expected) beacon will be issued before the line which waits for it? Can it happen?

Reply:
I have tried to explain the text context clearly and added additional test line to verify expected behaviour.